### ISSUE TITLE
? missing publications bug

### ISSUE THREAD

**Issue by jessflem** [opentrials/opentrials#514]
it looks like we've lost some publication links eg: [this trial](http://explorer.opentrials.net/trials/efd81fbe-91a8-4ac3-82f7-0d644d9ea715) has no linked publications, even though these publications [here](https://www.ncbi.nlm.nih.gov/pubmed/25582269?dopt=Abstract) and [here](https://www.ncbi.nlm.nih.gov/pubmed/19876678?dopt=Abstract) both have NCT ids in (and I assume we used the publications to match the cochrane data to the trial).

**Comment by georgiana-b** [comment]
It appears the only way the cochrane review could have been linked is through [this publication](https://www.ncbi.nlm.nih.gov/pubmed/20633377). This is the reference extracted from the XML:

```json
[
  {
    "no": "2",
    "pg": "106-11",
    "vl": "1",
    "type": "JOURNAL_ARTICLE",
    "year": "2008",
    "title": "Repetitive transcranial magnetic stimulation (rTMS) as an augmentation treatment for the negative symptoms of schizophrenia: A 4-week randomized placebo controlled study",
    "source": "Brain Stimulation",
    "authors": "Schneider A, Schneider T, Stark H",
    "country": "",
    "identifiers": []
  }
]
```
So, by following the algorithm, it is matched with [this publication from `opentrials`](https://query.opentrials.net/queries/133). A query [in Redash by this publication's id](https://query.opentrials.net/queries/134) shows that this publication is not linked to any trial anymore.
Since the review ended up connected to that trial, I assume at some point the above mentioned publication was linked (not necessarily *correctly*) to [this trial](https://explorer.opentrials.net/trials/efd81fbe-91a8-4ac3-82f7-0d644d9ea715).

So I think the issue is that when a link between a trial and a publication is removed, the review results should also be removed if they were linked to the trial thorough that publication.

@vitorbaptista To test my hypothesis, had links between `trials` and `publications` been removed on production since the `cochrane_reviews` has been run?

**Comment by benmeg** [comment]
A user just submitted another example:

Trial name: Magnetic Resonance Imaging in Women Receiving Chemotherapy for Stage III Breast Cancer
NCTID: NCT00043017

* OpenTrials (no publications): https://explorer.opentrials.net/trials/4276144f-4813-4a8a-9357-cf130c8c885a
* ClinicalTrials.gov (4 linked PubMed publications + 11 unlinked comprised of 4 journal article supplements and 7 symposium abstracts): https://clinicaltrials.gov/ct2/show/results/NCT00043017

A few observations:

* For this example, the publications appear in both the [HTML](https://clinicaltrials.gov/ct2/show/NCT00043017) and the [XML](https://clinicaltrials.gov/ct2/show/NCT00043017?displayxml=true), but in @jessflem's [above example](https://github.com/opentrials/opentrials/issues/514#issue-183992023), the publications are only present in the [HTML](https://clinicaltrials.gov/ct2/show/NCT00783120) (and not in the [XML](https://clinicaltrials.gov/ct2/show/NCT00783120?displayxml=true))
* For this example, the publications in the [XML](https://clinicaltrials.gov/ct2/show/NCT00043017?displayxml=true) are listed under the `<results_reference>` structure, and for some publications a PubMed ID is listed in the `<PMID>` field. All the article details (i.e. author list, article title, journal name, date, issue/page number, and DOI) are simply bunched together under the `<citation>` tag which isn't ideal, but we should be able to do a lookup using the PubMed ID if we need to extract structured data (or parse the `<citation>` field ourselves).
* **Possible solution:** For both examples, in the HTML, publications with a PubMed ID are linked to PubMed articles (via a ClinicalTrials.gov redirector link `href="/ct2/bye/`). They also have the CSS property `class='study-link'` which is not unique to just the publication links. However, the link redirector and CSS property only exist together for PubMed links. So, we could simply look for `<a class='study-link' href="/ct2/bye/` in the HTML of the ClinicalTrials.gov page and follow the redirect link to grab the associated PubMed article and link it to the corresponding OpenTrials trial page.

This section from ClinicalTrials.gov may be relevant - 'Find Studies With Results Published in a Medical Journal': https://www.clinicaltrials.gov/ct2/help/how-find/find-study-results#Published - I didn't realise that they do automatic matching by looking for the NCTID as we do (from a quick look it seems they also search at least the abstract and metadata).

**Comment by georgiana-b** [comment]
@benmeg This is a great catch! I opened #634 for it.

**Comment by nightsh** [comment]
@georgiana-b any more information to push this issue forward a bit? Is it blocked (or invalidated) by #634?

**Comment by georgiana-b** [comment]
@nightsh I posed a question in [my first comment](https://github.com/opentrials/opentrials/issues/514#issuecomment-257297227) that has not been answered. The original issue here is that we had `publications` that were linked with `trials` but for reasons still unknown the link was removed. 
#634 has to do with improving our `pubmed_linker` processor and was a side-effect of investigating this.

**Comment by pwalsh** [comment]
@georgiana-b @vitorbaptista can we close this issue and merge completely into #634 ?

**Comment by vitorbaptista** [comment]
Closing this, as it seems this could be solved via #634. Feel free to reopen if we're missing something.
